### PR TITLE
Move layout components out of layout directory

### PR DESCRIPTION
**Pull request by KyleAMathews**

### CODE REVIEW

**Comment by KyleAMathews** [comment]
https://github.com/gatsbyjs/gatsby/issues/1889 is the fix in Gatsby

File: www/src/layout-components/components/Header/Header.js [hunk 1]
```diff
@@ -48,12 +49,12 @@ const Header = ({location}) => (
             height: 40,
           },
         }}>
-        <a
+        <Link
```
Comment (KyleAMathews) [inline www/src/layout-components/components/Header/Header.js:52]: Fixed this while I was at it 😇

File: www/src/layouts/index.js [hunk 2]
```diff
@@ -13,13 +13,13 @@
 
 import React, {Component} from 'react';
 import docsearch from 'docsearch.js';
-import Flex from 'components/Flex';
-import Footer from './components/Footer';
-import Header from './components/Header';
+import Flex from '../components/Flex';
+import Footer from '../layout-components/components/Footer';
```
Comment (KyleAMathews) [inline www/src/layouts/index.js:17]: You'll probably want to either combine layout-components with the rest of the components or pick a better name for it.

Comment (bvaughn) [reply www/src/layouts/index.js:17]: Hm. Yeah. I initially started out with everything in a single `/components` directory but moved away from it b'c it felt too spread out.

**Review by bvaughn — APPROVED** [review]
Thanks @KyleAMathews!

**Pull request merged**